src/main/features/response/form/models/statement-of-means/monthlyIncome.ts

```diff
@@ -172,4 +172,9 @@ export class MonthlyIncome {
     this.otherSources.splice(this.otherSources.findIndex(element => element === source), 1)
   }
 
+  resetIncome (propertyName: string, source: IncomeSource): void {
+    this[`${propertyName.split('.')[0]}Declared`] = false
```

Comment:
Fully understand why we have this here. Maybe a bit much responsibility for the model. Wonder if there would be an easy enough way to doing the `propertyName` manipulation outside of the model?

Reply:
I take your point but I think it's ok here, its encapsulated in resetting a field, which the model should know best how to do.

unfortunately there's the complexity of there being two fields that need clearing